pkg/resources/istioingress/virtualservice.go

```diff
@@ -28,16 +28,72 @@ import (
 )
 
 func (r *Reconciler) virtualService(log logr.Logger, externalListenerConfig v1beta1.ExternalListenerConfig) runtime.Object {
+	vServiceSpec := v1alpha3.VirtualServiceSpec{
+		Hosts:    []string{"*"},
+		Gateways: []string{fmt.Sprintf(gatewayNameTemplate, r.KafkaCluster.Name, externalListenerConfig.Name)},
+	}
+
+	if r.KafkaCluster.Spec.IstioIngressConfig.TLSOptions != nil &&
+		r.KafkaCluster.Spec.IstioIngressConfig.TLSOptions.Mode == v1alpha3.TLSModePassThrough {
+		vServiceSpec.TLS = generateTLSRoutes(r.KafkaCluster, externalListenerConfig, log)
+
+	} else {
+		vServiceSpec.TCP = generateTcpRoutes(r.KafkaCluster, externalListenerConfig, log)
```

Comment:
generateTcpRoutes and generateTLSRoutes should follow the same naming conventions